src/config/entities/schemas/configuration.schema.ts

```diff
@@ -0,0 +1,61 @@
+import { z } from 'zod';
+
+export const RootConfigurationSchema = z
+  .object({
+    AUTH_TOKEN: z.string(),
+    AWS_ACCESS_KEY_ID: z.string().optional(),
+    AWS_SECRET_ACCESS_KEY: z.string().optional(),
+    AWS_REGION: z.string().optional(),
+    CGW_ENV: z.string().optional(),
+    LOG_LEVEL: z
+      .enum(['error', 'warn', 'info', 'http', 'verbose', 'debug', 'silly'])
+      .optional(),
+    // TODO: Reassess EMAIL_ keys after email integration
+    EMAIL_API_APPLICATION_CODE: z.string(),
+    EMAIL_API_FROM_EMAIL: z.string().email(),
+    EMAIL_API_KEY: z.string(),
+    EMAIL_TEMPLATE_RECOVERY_TX: z.string(),
+    EMAIL_TEMPLATE_UNKNOWN_RECOVERY_TX: z.string(),
+    EMAIL_TEMPLATE_VERIFICATION_CODE: z.string(),
+    INFURA_API_KEY: z.string(),
+    PUSH_NOTIFICATIONS_API_PROJECT: z.string(),
+    PUSH_NOTIFICATIONS_API_SERVICE_ACCOUNT_CLIENT_EMAIL: z.string().email(),
+    PUSH_NOTIFICATIONS_API_SERVICE_ACCOUNT_PRIVATE_KEY: z.string(),
+    RELAY_PROVIDER_API_KEY_OPTIMISM: z.string(),
+    RELAY_PROVIDER_API_KEY_BSC: z.string(),
+    RELAY_PROVIDER_API_KEY_GNOSIS_CHAIN: z.string(),
+    RELAY_PROVIDER_API_KEY_POLYGON: z.string(),
+    RELAY_PROVIDER_API_KEY_POLYGON_ZKEVM: z.string(),
+    RELAY_PROVIDER_API_KEY_BASE: z.string(),
+    RELAY_PROVIDER_API_KEY_ARBITRUM_ONE: z.string(),
+    RELAY_PROVIDER_API_KEY_AVALANCHE: z.string(),
+    RELAY_PROVIDER_API_KEY_LINEA: z.string(),
+    RELAY_PROVIDER_API_KEY_BLAST: z.string(),
+    RELAY_PROVIDER_API_KEY_SEPOLIA: z.string(),
+    STAKING_API_KEY: z.string(),
+    STAKING_TESTNET_API_KEY: z.string(),
+    TARGETED_MESSAGING_FILE_STORAGE_TYPE: z.enum(['local', 'aws']).optional(),
+  })
+  .superRefine((config, ctx) =>
+    // Check for AWS_* fields in production and staging environments
+    ['AWS_ACCESS_KEY_ID', 'AWS_SECRET_ACCESS_KEY', 'AWS_REGION'].forEach(
+      (field) => {
+        if (
+          config.CGW_ENV &&
+          config instanceof Object &&
+          ['production', 'staging'].includes(config.CGW_ENV) &&
+          !(config as Record<string, unknown>)[field]
+        ) {
+          ctx.addIssue({
+            code: z.ZodIssueCode.custom,
+            message: `is required in production and staging environments`,
+            path: [field],
+          });
+        }
+      },
+    ),
```

Comment:
Why do we need to `superRefine` this? We could just remove `optional()` from the respective property.

Reply:
I thought it would be nice to avoid enforcing these on the `env.sample` file and in the local development environments, as we have a `local` FileStorageType, the need to set those `AWS_*` env variables when the `local` FileStorageType is selected would be a bit weird IMO. 

Using this `superRefine` approach, we could even apply it to other environment variables, to stop requiring placeholders in development environments and simplify the `safe-infrastrastructure` repository management. Wdyt? (Of course, I'd tackle that in a separate PR if you agree).